tests/func/test_status.py

```diff
@@ -183,3 +184,42 @@ def test_params_without_targets(tmp_dir, dvc):
             }
         ]
     }
+
+
+def test_dir_optimization(tmp_dir, dvc, mocker):
```

Comment:
Would it make sense to test `data.stage` instead?

Reply:
Moved the test to `test_stage.py` and adapted it